src/shims/unix/socket.rs

```diff
@@ -42,17 +105,40 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
     ) -> InterpResult<'tcx, Scalar<Provenance>> {
         let this = self.eval_context_mut();
 
-        let _domain = this.read_scalar(domain)?.to_i32()?;
-        let _type_ = this.read_scalar(type_)?.to_i32()?;
-        let _protocol = this.read_scalar(protocol)?.to_i32()?;
+        let domain = this.read_scalar(domain)?.to_i32()?;
+        let type_ = this.read_scalar(type_)?.to_i32()?;
+        let protocol = this.read_scalar(protocol)?.to_i32()?;
         let sv = this.deref_pointer(sv)?;
 
-        // FIXME: fail on unsupported inputs
+        // Fail on unsupported input
+        if domain != this.eval_libc_i32("AF_UNIX") && domain != this.eval_libc_i32("AF_LOCAL") {
+            throw_unsup_format!("socketpair: Only AF_UNIX and AF_LOCAL are supported");
+        } else if type_ != this.eval_libc_i32("SOCK_STREAM")
+            && type_ != this.eval_libc_i32("SOCK_DGRAM")
+        {
+            throw_unsup_format!("socketpair: Only SOCK_STREAM and SOCK_DGRAM are suppported");
+        } else if protocol != 0 {
+            throw_unsup_format!("socketpair: Unsupported socket protocol is used");
```

Comment:
report the protocol id in the message (same for `domain` and `type_`)

Reply:
I wonder if we should report the exact flag name, because when I did the test 
```rust
    let flag = libc::SOCK_STREAM | libc::SOCK_NONBLOCK | libc::SOCK_CLOEXEC;
    let mut fds = [-1, -1];
    let mut res = unsafe { libc::socketpair(libc::AF_UNIX, flag, 0, fds.as_mut_ptr()) };
```
and the error shown is not very descriptive
```bash
error: unsupported operation: socketpair: Unsupported type 526337 is used
   --> ./tests/pass-dep/libc/libc-socketpair.rs:104:28
    |
104 | ... { libc::socketpair(libc::AF_UNIX, flag, 0, fds.as_mut_ptr()) };
    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ socketpair: Unsupported type 526337 is used
```

Reply:
For ``socketpair``, it is still manageable to parse the flags and report corresponding unsup error for each flag, but for other bigger functions like ``epoll`` that can have up to ``12`` flags, I am not sure if it is still a good idea to do this.

Reply:
I think it's ok to just report the integer. It can be looked up manually. If there's an easy way to describe the value, we can use that, but otherwise any value is better than none in my book

Reply:
In my experience, it is very very rare to be familiar with the integer values, because they are not specified. People learn the names because that is what is portable.

In an effort to be helpful, can we mention _is_ allowed?

Reply:
It helps a bit to print the integer value in hex, since there's usually bitwise combination of flags involved.

But decoding the flag and showing it with the names is quite a bit of work and requires knowing the names and values of all the relevant bits. If we want to do that I suggest it should be a helper function we can use in multiple places, and a separate PR.

Reply:
Combining all the ideas, we can do print value in hex + mention what is allowed, for example:

```rust
throw_unsup_format!("socketpair: type {:x} is not supported, only SOCK_STREAM, SOCK_CLOEXEC and SOCK_NONBLOCK are allowed", _type);
```

It's a bit lengthy, but this is the longest that it can get for ``socketpair``

Reply:
Sounds good, except please add the leading `0x` to the hex output. (`{:#x}` will do that).